### PR TITLE
Added plugins

### PR DESCRIPTION
**Pull request by DJuan33**
Added 4 new plugins:
 - **LSPsaga:** https://github.com/glepnir/lspsaga.nvim
 - **Hop:** https://github.com/phaazon/hop.nvim
 - **Trouble:** https://github.com/folke/trouble.nvim
 - **Todo Comments:** https://github.com/folke/todo-comments.nvim
 
 All plugins have lazy loading, .chadrc options, mappings and init settings. All plugins was tested by me.

### CODE REVIEW

File: lua/pluginList.lua [hunk 1]
```diff
@@ -324,4 +324,38 @@ return packer.startup(function()
...
+     after = "nvim-lspconfig",
+     config = function ()
+       require("plugins.others").saga()
+     end,
+     setup = function ()
+       require("mappings").saga()
+     end
+   }
+
+  use {
```
Comment (G-Rowell) [inline lua/pluginList.lua:339]: It looks like `Hop` has some notable issues with the current NeoVim stable release,
advanced Vim motions are really cool and I like the idea.....but I don't think `Hop` is ready for use due to bugs.

[See first paragraphs of it's README](https://github.com/phaazon/hop.nvim#readme)

`it suffers from a few drawbacks making it not comfortable to use with Neovim version >0.5 – at least at the time of writing these lines:`

**Comment by siduck76** [comment]
Bruhhh these many plugins , noooooo :c

**Comment by siduck76** [comment]
Please read the contribute section again

File: lua/chadrc.lua [hunk 2]
```diff
@@ -126,6 +126,31 @@ local M = {
...
+
+      },
+      hop = {
+        hop_line = "<leader>pl",
+        hop_word = "<leader>pw",
+        hop_char1 = "<leader>p1",
+        hop_char2 = "<leader>p2",
+
+      },
+      trouble = {
```
Comment (G-Rowell) [inline lua/chadrc.lua:149]: I think Trouble provides some potentially nice functionality,
though I do think that this the user would have their own opinions as to if they want it or not.

Definitely something that should have an `enable/disable` option in `chadrc.lua`, and we should move to prompting what the user wants enabled/disabled on `install.sh`

File: lua/pluginList.lua [hunk 3]
```diff
@@ -324,4 +324,38 @@ return packer.startup(function()
...
+     "phaazon/hop.nvim",
+     event = "BufRead",
+     cmd = {"HopWord", "HopLine", "HopChar1", "HopChar2", "HopPattern"},
+     config = function() require("plugins.others").hop() end,
+     setup = function() require("mappings").hop() end
+  }
+
+  use {
```
Comment (G-Rowell) [inline lua/pluginList.lua:347]: Hmmmm, 
I'm worried that this plugin may be a little small in the functionality provided, especially given `Treesitter/LSP` highlight TODOs already. 

Would you elaborate on what this plugin provides? (I've read through it's Readme, but maybe I'm missing something)

File: lua/pluginList.lua [hunk 4]
```diff
@@ -324,4 +324,38 @@ return packer.startup(function()
          require("mappings").fugitive()
       end,
    }
+
+   use {
```
Comment (G-Rowell) [inline lua/pluginList.lua:328]: Hmmmm, I do think we may be missing some of the functionality of LSPSaga?

Like autodoc hover, I'm not sure if anything we have does this already?

**Review by G-Rowell — CHANGES_REQUESTED** [review]
Think we may need to reconsider 1 or two of these plugins, 
feel free to describe why you think they are important ofc!

**Comment by G-Rowell** [comment]
@DJuan33 

Thank you for your first PR to NvChad!!!
We really appreciate your effort, let's see what we can do

**Comment by siduck76** [comment]
@DJuan33 we may accept the PR for trouble.nvim only and for v1.5 , but now now :(

**Comment by G-Rowell** [comment]
image

@DJuan33 check out `:Telescope lsp_document_diagnostics` when you have the appropriate `:LspInstall <language` if you haven't already

**Comment by G-Rowell** [comment]
Hi @DJuan33,

I think we might close this, as two of the plugins have their functionality completely provided by `Telescope` & `:lsp_workspace_diagnostics` & `:lsp_document_diagnostics`.

Furthermore `Hop` doesn't seem stable with `NeoVim 0.5+`.

We all really appreciate your effort and would encourage you to contribute again!
If you have any questions at all, we're happy to answer them and we're still open to suggestions & discussion around any PR.

G

**Comment by DJuan33** [comment]
Haha well, I see in the Issues and Reddit that LSPSaga is popular and requested. Maybe a good plugin to use with Signature. I'm agree about Hop plugin is situational and I didn't know Treesitter highlight the TODOs already. However, I think trouble is useful. 

I'm not familiariced with Lua and Neovim API, but I like to contribute. Thanks for the comments.

**Pull request closed without merging**